src/utils/upcoming-matches.ts

```diff
@@ -0,0 +1,19 @@
+import { ProcessedMatchInfo } from '@/api/match-info'
+// import { useCurrentTime } from './use-current-time'
+
+export const getUpcomingMatches = (
+  matches: ProcessedMatchInfo[],
+  currentTime: number,
+) => {
+  const retArr = []
+  for (const match of matches) {
+    if (
+      match.time &&
+      match.time.getTime() < currentTime + 1000 * 60 * 15 &&
+      match.time.getTime() > currentTime - 1000 * 60 * 4
```

Comment:
@varCepheid It looks like this only shows matches between 4 minutes ago and 15 minutes from now.

I am looking at the madtown throwdown event that is currently happening. Here's what it looks like on `main`:

https://peregrinefrc.com/events/2023mttd/teams/1678

![image](https://github.com/Pigmice2733/peregrine-frontend/assets/13206945/0b70361a-c727-4a09-920c-911236208c0e)

This is helpful to me because it shows when the team will compete next, even though it is more than 15 minutes from now.

On this branch, here is what it shows:

![image](https://github.com/Pigmice2733/peregrine-frontend/assets/13206945/70a90509-730f-47f1-97cf-b21e08f11b2b)

The "no upcoming matches" text is a little confusing to me since it would make me assume that they do not have any matches coming up, while in reality they are probably queueing for the match (the time of the screenshot is 9:06 AM)

Another potential problem/improvement that we could implement is that matches that already have scores do not need to be shown in "upcoming matches" even if they were less than 4 minutes ago (since we know the match already finished). You can copy this logic from the next-incomplete-match file that was deleted.

Can I propose this logic for getUpcomingMatches instead?

```js
const ONE_DAY = 24 * 60 * 60 * 1000;
return matches
  // Only include matches that haven't finished yet (scores haven't been posted yet)
  .filter(match => match.redScore === undefined)
  // If 24h has passed since a match's predicted time without scores being posted,
  // it probably won't ever be played.
  // For example: tiebreaker matches that won't get played
  .filter(match.time.getTime() > currentTime - ONE_DAY)
  // limit to showing 3 matches
  .slice(0, 3)
```

This way, we can avoid all timing-based logic, and that avoids a bunch of time-related issues like field delays right before a match, and it still shows matches for the next day that teams will play in, which is nice.

It also would avoid showing for too long tiebreaker matches that never will be played, like https://www.thebluealliance.com/match/2023brd_f1m3 (since the red alliance won F1 and F2, the F3 tiebreaker does not need to be played, but the FMS never went and deleted that match)